ironfish/src/wallet/walletdb/walletdb.test.ts

```diff
@@ -47,6 +55,126 @@ describe('WalletDB', () => {
     })
   })
 
+  describe('loadUnspentNotesByValue', () => {
+    it('notes should be returned in order of value', async () => {
+      const { node } = await nodeTest.createSetup()
+      const account = await useAccountFixture(node.wallet)
+
+      const block1 = await useMinerBlockFixture(node.chain, undefined, account)
+      await expect(node.chain).toAddBlock(block1)
+      await node.wallet.updateHead()
+
+      for (let i = 1; i < 3; i++) {
+        const transaction = await useTxFixture(node.wallet, account, account)
+        await expect(node.chain).toAddBlock(
+          await useMinerBlockFixture(node.chain, undefined, undefined, undefined, [
+            transaction,
+          ]),
+        )
+        await node.wallet.updateHead()
+      }
+
+      const walletDb = node.wallet.walletDb
+      const noteHashes = await AsyncUtils.materialize(
+        walletDb.loadValueToUnspentNoteHashes(account, Asset.nativeId()),
+      )
+      const unspentNotes = await AsyncUtils.materialize(
+        walletDb.loadUnspentNoteHashes(account, Asset.nativeId()),
+      )
+      const notes = (
+        await Promise.all(
+          noteHashes.map((noteHash) => walletDb.loadDecryptedNote(account, noteHash)),
+        )
+      ).filter((note) => note !== undefined) as DecryptedNoteValue[]
+
+      expect(notes).toHaveLength(unspentNotes.length)
+
+      let previousNoteValue = notes[0].note.value()
+
+      for (const note of notes) {
+        expect(note.note.value()).toBeGreaterThanOrEqual(previousNoteValue)
+        previousNoteValue = note.note.value()
+      }
+    })
+
+    it('keys should be stored in order', async () => {
+      const { node } = await nodeTest.createSetup()
+      const walletDb = node.wallet.walletDb
+      const account = await useAccountFixture(node.wallet)
+
+      await walletDb.valueToUnspentNoteHashes.put(
+        [account.prefix, [Asset.nativeId(), [100n, Buffer.from('100', 'hex')]]],
+        null,
+      )
+      await walletDb.valueToUnspentNoteHashes.put(
+        [account.prefix, [Asset.nativeId(), [1n, Buffer.from('1', 'hex')]]],
+        null,
+      )
+      await walletDb.valueToUnspentNoteHashes.put(
+        [account.prefix, [Asset.nativeId(), [10n, Buffer.from('10', 'hex')]]],
+        null,
+      )
+      await walletDb.valueToUnspentNoteHashes.put(
+        [account.prefix, [Asset.nativeId(), [1000n, Buffer.from('1000', 'hex')]]],
+        null,
+      )
+      await walletDb.valueToUnspentNoteHashes.put(
+        [account.prefix, [Asset.nativeId(), [10000n, Buffer.from('10000', 'hex')]]],
+        null,
+      )
+
+      const allNotes = await walletDb.valueToUnspentNoteHashes.getAll()
+
+      const correctOrder = [1n, 10n, 100n, 1000n, 10000n]
+
+      const values = allNotes.map((note) => note[0][1][1][0])
+
+      expect(values).toEqual(correctOrder)
```

Comment:
This could be a bit more concise and still pretty clean I think

```suggestion
      const { wallet } = await nodeTest.createSetup()
      const account = await useAccountFixture(wallet)

      const values = [100n, 1n, 10n, 1000n, 10000n]
      const expected = [1n, 10n, 100n, 1000n, 10000n]

      await Promise.all(
        values.map((value) =>
          wallet.walletDb.valueToUnspentNoteHashes.put(
            [account.prefix, [Asset.nativeId(), [value, Buffer.alloc(32, Math.random())]]],
            null,
          ),
        ),
      )

      const allNotes = await wallet.walletDb.valueToUnspentNoteHashes.getAll()
      expect(allNotes.map((index) => index[0][1][1][0])).toEqual(expected)
```

Reply:
Done!